server/src/main/java/org/opensearch/index/IndexingPressureService.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index;
+
+import org.opensearch.action.admin.indices.stats.CommonStatsFlags;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.lease.Releasable;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.index.stats.IndexingPressureStats;
+import org.opensearch.index.stats.ShardIndexingPressureStats;
+
+/**
+ * Sets up classes for node/shard level indexing pressure.
+ * Provides abstraction and orchestration for indexing pressure interfaces when called from Transport Actions or for Stats.
+ */
+public class IndexingPressureService {
+
+    private final ShardIndexingPressure shardIndexingPressure;
+
+    public IndexingPressureService(Settings settings, ClusterService clusterService) {
+        shardIndexingPressure = new ShardIndexingPressure(settings, clusterService);
+    }
+
+    public Releasable markCoordinatingOperationStarted(ShardId shardId, long bytes, boolean forceExecution) {
+        if (isShardIndexingPressureEnabled()) {
+            return shardIndexingPressure.markCoordinatingOperationStarted(shardId, bytes, forceExecution);
+        } else {
+            return shardIndexingPressure.markCoordinatingOperationStarted(bytes, forceExecution);
+        }
+    }
```

Comment:
Instead of if-else block it'd be much better if we had the same downstream call which based on settings in their respective classes could have handled these cases

Reply:
Agree thats one way of doing this. Wanted the Service to take care of orchestration logic, as it simplifies addition of new similar constructs in future and even helps deprecation of the older constructs faster. We are also exposing dedicated plumbing logic for callers (such as part of #1113) which routes request for the ease of callers. Let me know if you strongly feel otherwise.